builder/src/lib.rs

```diff
@@ -365,10 +365,12 @@ fn build_delta(rootfs: &Path, oci: &Image, mut existing: Option<PuzzleFS>) -> Re
     // TODO: not render this whole thing in memory, stick it all in the same blob, etc.
     let mut dir_buf = Vec::<u8>::new();
 
+    let mut sorted_dirs = dirs.values_mut().collect::<Vec<_>>();
```

Comment:
Hm, now, what dictates the inode numbers here?  Is it strictly the order in which files are presented to 'puzzlefs build'?

Reply:
(This is just a question out of my curiousity.  Merging as it looks good, and is passing the new tests :) )

Reply:
Good question, inode numbers are allocated sequentially, based on the order in which puzzlefs traverses the directory:
* https://github.com/anuvu/puzzlefs/blob/master/builder/src/lib.rs#L25 here it sorts the entries by name
* https://github.com/anuvu/puzzlefs/blob/master/builder/src/lib.rs#L215 here it uses read_dir, which doesn't sort the entries in any way + the order is filesystem dependent

I will reopen #43 and create a new PR in which I'll sort the entries returned by read_dir.